### PR TITLE
Version relaxation + Optional inference model

### PR DESCRIPTION
**Pull request by oliveradk**
changed all version requirements to >=, added ability to toggle inference mode on statistical detector (apologies for multi-issue pr!)

### CODE REVIEW

**Comment by ejnnr** [comment]
Thanks! What's the use case for toggling on/off inference mode? I could imagine statistical detectors that want to collect gradients, just wondering whether this should be an init argument or just a detector property (and also how we should treat it if we refactor this at some point)

**Comment by oliveradk** [comment]
> Thanks! What's the use case for toggling on/off inference mode? I could imagine statistical detectors that want to collect gradients, just wondering whether this should be an init argument or just a detector property (and also how we should treat it if we refactor this at some point)

tentatively think it should be changed to torch.no_grad, and then detectors can overwrite with torch.with_grad() (similar to how eval works)
